lib/elixir/src/elixir_parser.yrl

```diff
@@ -659,9 +671,9 @@ meta_from_token_with_end_line(Begin, End) ->
   case ?formatter_metadata() of
     true ->
       [{end_line, line_from_location(?location(End))}
-       | meta_from_token(Begin)];
+       | meta_from_token_keep_eol(Begin)];
     false ->
-      meta_from_token(Begin)
+      meta_from_token_keep_eol(Begin)
```

Comment:
Instead of changing this function to also handle `eol`, what if we change this line:

https://github.com/Sihui/elixir/blob/d9252a8872667e2fd8b77dd78e50667761838c4f/lib/elixir/src/elixir_parser.yrl#L849

To do something like `eol_pair(Fn, End) ++ meta_from_token(Fn)` instead of `meta_from_token_with_end_line(Fn, End)`? I think we will end-up with less code after all.

Btw, great solution to this problem!

Reply:
@josevalim, good call! I agree. Using `meta_from_token_keep_eol` is not very optimal. Updated the PR accordingly :D